framework/sandstone.cpp

```diff
@@ -2039,7 +2030,18 @@ static void run_one_test_children(ChildrenList &children, int *tc, const struct
 static TestResult run_one_test_once(int *tc, const struct test *test)
 {
     ChildrenList children;
-    run_one_test_children(children, tc, test);
+    if (uint64_t missing = (test->minimum_cpu | test->compiler_minimum_cpu) & ~cpu_features) {
+        init_per_thread_data();
+
+        // for brevity, don't report the bits that the framework itself needs
+        missing &= ~_compilerCpuFeatures;
+        log_skip(CpuNotSupportedSkipCategory, "test requires %s\n", cpu_features_to_string(missing).c_str());
+        (void) missing;
```

Comment:
Why is this needed?..

Reply:
What "this" are you referring to? 
a) The moving up into the parent process? See commit message
b) the splitting into two functions? To keep Git Blame more or less intact
c) the `(void) missing` line? To avoid a warning about set-but-unused variable under `NO_LOGGING` mode.